sdk/storage/azdatalake/filesystem/client.go

```diff
@@ -202,7 +203,15 @@ func (fs *Client) BlobURL() string {
 // The new directory.Client uses the same request policy pipeline as the Client.
 func (fs *Client) NewDirectoryClient(directoryPath string) *directory.Client {
 	directoryPath = strings.ReplaceAll(directoryPath, "\\", "/")
-	dirURL := runtime.JoinPaths(fs.generatedFSClientWithDFS().Endpoint(), directoryPath)
+	names := strings.Split(directoryPath, "/")
+	path := ""
+	for i, name := range names {
```

Comment:
There is a sub-directory client also, just check if changes are needed there as well or not

Reply:
Subdirectory client takes name of the directory instead of the path, so changes are not needed there